### ISSUE TITLE
Reproducing Results for Figure2

### ISSUE THREAD

**Issue by won-bae** [yangalan123/anhp-andtt#2]
Hi authors, 
Thank you for sharing the great work along with fixed versions for the previous works: THP and SAHP.
While I was reproducing the results for THP, I ran into a problem.

- NLL: I was able to reproduce the results on stackoverflow but the results on MIMIC easily go down to ~5, which is far below than the reported number ~8. I have only changed lr from 1e-4 to 1e-3. Is it expected or am I missing something here?
- RMSE: it seems like the code for RMSE is not fixed the current version of the code. I believe all the dummy values for `event_time` and `prediction` need to be ruled out here:
https://github.com/yangalan123/anhp-andtt/blob/b907f3808ed2ce1616edb1bc2229993a6742cee9/thp/thp_training/Utils.py#L103
Could you confirm that it has not been fixed yet? If so, do you have a plan to fix it anytime soon?

Lastly, according to the datasets in google drive, each dataset is divided into several `fold`. Have you used all the folds together to report the numbers in Figure2? Or used a particular fold?
Thank you in advance.

**Comment by yangalan123** [comment]
Hi Won-bae,
Thanks for your interest in our works. Regarding the fix for the previous work THP and SAHP, let me explain the details here:

1. NLL: It is possible that you may obtain better results by playing with hyperparameter tuning. In our work, we provide a code fix for THP. After discussion with THP authors, we agree to only provide the code fix when codes are released and keep other parts as-is. The hyperparameter search for THP, by agreement, goes beyond our work. 

2. For RMSE, we have not uploaded our fixed prediction codes. As we write in [README](https://github.com/yangalan123/anhp-andtt/tree/master/thp#package-description), the RMSE values in the current `thp_training` are intentionally kept dummy because principally we need to use the thinning algorithm (You can check Appendix E in our paper) to do prediction, which is missing for now. I would upload the prediction part later. 

For Figure 2 in the paper, we use fold-1 to report. 

Best,
Chenghao Yang

**Comment by won-bae** [comment]
Hi Alan,
Thank you for the clarification. As a followup question for 2, I guess the problem of THP is that it reported the numbers based on the predictions from additional time and type prediction heads, which does not require to do predictions using thinning algorithm.
Given that THP specified that using additional time and type prediction heads gives better performance in sec 5.4, was there any particular reason that you decided to take them out and just use intensity predictions?

**Comment by yangalan123** [comment]
Hi Won-bae,
Sorry for the late reply as I was busy with some company works. Let me explain why we need to drop the usage of type & time prediction head and use a thinning algorithm (which only needs intensity predictions):

If we use type & time prediction head, then we are actually not building a true **generative model** (a probability distribution for continuous-time event streams), we just train an embedding model (a "feature extractor") with **generative loss** and **discriminative loss** together, and the embedding is used to predict the type and time.  However, as our focus is to build a **generative model**, the principal way to make predictions based on continuous-time generative model is thinning algorithm. We drop those prediction heads and write the thinning algorithms for THP and SAHP mainly for a fair comparison. 

Let me know if there is something not clear. Thanks!

**Comment by JardinDelSol** [comment]
Hi Alan,

First of all, I also would like to thank you for sharing a great work.

I was wondering if [this function(draw_next_time_ordinary)](https://github.com/yangalan123/anhp-andtt/blob/b907f3808ed2ce1616edb1bc2229993a6742cee9/anhp/esm/thinning.py#L46) was the code that was used to produce the prediction results in the paper.

If so, I'm having some hard time understanding the notion that the implemented thinning algorithm can "sample t_i drawn from p_i(t)" as you mentioned in section E.
I would really appreciate it if you could provide me some materials that I can work with to understand the mathmetics behind it.

Thank you.

**Comment by yangalan123** [comment]
Hi JardinDelSol @JardinDelSol ,
Thanks for your interest in my codes! For thinning algorithm explanation, you are right in that I do use [draw_next_time_ordinary()](https://github.com/yangalan123/anhp-andtt/blob/b907f3808ed2ce1616edb1bc2229993a6742cee9/anhp/esm/thinning.py#L46) to make the prediction. I can surely explain to you those mathematical details, but I was wondering whether you could open up another issue since this seems an orthogonal question to @won-bae, and another issue can also help make our discussion focused. 

I totally understand that the thinning algorithm is a bit hard to understand here. I would recommend reading Section B.3 in Appendix of the [paper](https://arxiv.org/pdf/1612.09328.pdf), which has a more clear discussion and explanation of the thinning algorithm.

**Comment by yangalan123** [comment]
Close this issue as no further response is observed. Feel free to re-open it if you have any further questions. Thanks!
